File: back-side/persistence/src/main/java/esp/lord/dic1/uml/persistence/medicalfile/entities/ConsultationSheet.java
package esp.lord.dic1.uml.persistence.medicalfile.entities;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;

@NoArgsConstructor @ToString @Setter @Getter
@Entity
public class ConsultationSheet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer sheetNumber;
    Date creationDate;
    Date modificationDate;
    String notes;
    @ManyToOne
    MedicalFile medicalFile;
    @OneToMany(fetch = FetchType.LAZY)
    List<Analysis> analyzes;
}

File: front-side/webappts/src/types/consultationsheet.ts
export interface ConsultationSheet {
    sheetNumber: number,
    creationDate: Date | string,
    modificationDate: Date | string,
    notes: string
}

export interface ConsultationSheetFields  {
    modificationDate:string,
    creationDate?:string,
    sheetNumber?: number,
    notes?: string
}
